### PR TITLE
Add a uint indexer

### PR DESCRIPTION
**Pull request by dadgar**
This PR introduces a uint indexer that works for any uint type: uint,
uint8, uint16, uint32, uint64.

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -249,6 +250,66 @@ func (s *StringMapFieldIndex) FromArgs(args ...interface{}) ([]byte, error) {
...
+	if !v.IsValid() {
+		return nil, fmt.Errorf("%#v is invalid", args[0])
+	}
+
+	if k := v.Kind(); !IsUintType(k) {
+		return nil, fmt.Errorf("arg is of type %v; want a uint", k)
+	}
+
+	val := v.Uint()
+	buf := make([]byte, 8)
```
Comment (slackpad) [inline index.go]: You think it's worth exploiting a memory footprint advantage by having `IsUintType()` return the required size and allocating just what's needed?

Comment (slackpad) [inline index.go]: Probably not worth the complexity...

Comment (dadgar) [inline index.go]: It was easy enough! Done!

**Comment by fta1** [comment]
linking https://github.com/hashicorp/go-memdb/issues/30 related to this PR

**Pull request merged**